lib/decimal.ex

```diff
@@ -267,8 +267,16 @@ defmodule Decimal do
     num2
   end
 
+  def add(%Decimal{coef: :inf, sign: sign, exp: exp1} = num1, %Decimal{coef: :inf, sign: sign, exp: exp2}) do
+    if exp1 > exp2 do
+      num1
+    else
+      num2
```

Comment:
You need to add `= num2` to the function or implement the condition as a function guard with two clauses.